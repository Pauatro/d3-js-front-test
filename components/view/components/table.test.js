import createViewContainer from './view-container.js'
import createTable from './table.js'

it('renders correctly', () => {

    const container = document.createElement('div')
    container.style = "width: 300px; height: 200px;"

    document.body.append(container)

    const {
        svg,
        containerHeight,
        containerWidth
    } = createViewContainer(container, 0)

    const outerRadius = containerHeight/3.5
    const innerRadius = outerRadius*0.9

    const grouped = [
        {
            "group": "tablet",
            "perc": 60,
            "total": 120000
        },
        {
            "group": "smartphone",
            "perc": 40,
            "total": 80000
        }
    ]

    createTable(svg, containerWidth, containerHeight, innerRadius, outerRadius, grouped, 'â‚¬', 0)
    expect(svg).toMatchSnapshot();
});