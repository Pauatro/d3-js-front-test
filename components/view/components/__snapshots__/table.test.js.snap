// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Selection {
  "_groups": Array [
    Array [
      <svg
        viewBox="0,0,0,0"
      >
        <foreignobject
          height="0"
          transform="translate(0, 0)"
          width="0"
        >
          <div
            class="table-container"
            style="
            width: 0px;
            height: 0px;
            display: flex;
            align-items: flex-start;
            justify-content: center;
        "
          >
            <table
              style="
            width: 0px;
        "
            >
              <thead
                class="table-header"
              >
                <tr>
                  <th
                    style="
            width: 50%;
            text-align: left;
            color: #3a6815;
            font-size: 0px;
        "
                  >
                    Tablet
                  </th>
                  <th
                    style="
            width: 50%;
            text-align: right;
            color: #93d55d;
            font-size: 0px;
        "
                  >
                    Smartphone
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    align="left"
                    class="table-body"
                    style="
            width: 50%;
            padding-top: 0px;
            padding-bottom: 0px;
            margin: 0px;
            border-spacing: 0px;
        "
                  >
                    <td
                      align="left"
                      class="table-body-left"
                      style=" 
            font-size: 0px; 
            padding-right: 10px;
        "
                    >
                      60%
                    </td>
                    <td
                      align="left"
                      class="table-body-right"
                      style=" 
            font-size: 0px; 
            color: #b3b3b3;        
        "
                    >
                      120.000€
                    </td>
                  </td>
                  <td
                    align="right"
                    class="table-body"
                    style="
            width: 50%;
            padding-top: 0px;
            padding-bottom: 0px;
            margin: 0px;
            border-spacing: 0px;
        "
                  >
                    <td
                      align="right"
                      class="table-body-left"
                      style=" 
            font-size: 0px; 
            padding-right: 10px;
        "
                    >
                      40%
                    </td>
                    <td
                      align="right"
                      class="table-body-right"
                      style=" 
            font-size: 0px; 
            color: #b3b3b3;        
        "
                    >
                      80.000€
                    </td>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </foreignobject>
      </svg>,
    ],
  ],
  "_parents": Array [
    null,
  ],
}
`;
